pygis/docs/e_summarize_vector.md

```diff
@@ -0,0 +1,569 @@
+
+For further reading, check out the [`scipy` documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.gaussian_kde.html).
+
+```{code-cell} ipython3
+# Code adapted from https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.gaussian_kde.html
+
+# Get X and Y coordinates of well points
+x_sp = wells["geometry"].x
+y_sp = wells["geometry"].y
+
+# Get minimum and maximum coordinate values of well points
+min_x_sp, min_y_sp, max_x_sp, max_y_sp = wells.total_bounds
+
+# Create a 100 by 100 cell mesh grid
+# Horizontal and vertical cell count should be the same
+XX_sp, YY_sp = np.mgrid[min_x_sp:max_x_sp:100j, min_y_sp:max_y_sp:100j]
+
+# Create 2-D array of the coordinates (paired) of each cell in the mesh grid
+positions_sp = np.vstack([XX_sp.ravel(), YY_sp.ravel()])
+
+# Create 2-D array of the coordinate values of the well points
+values_sp = np.vstack([x_sp, y_sp])
+
+# Create kernel density estimator using the values
+kde_sp = stats.gaussian_kde(values_sp)
+
+# Set kernel bandwidth (optional)
+kde_sp.set_bandwidth(0.1)
+
+# Evaluate the estimator and reshape the data to fit mesh grid
+Z_sp = np.reshape(kde_sp.evaluate(positions_sp), XX_sp.shape)
+
+# Plot data
+fig, ax = plt.subplots(1, 1, figsize = (10, 10))
+ax.imshow(np.rot90(Z_sp), cmap = "RdPu", extent = [min_x_sp, max_x_sp, min_y_sp, max_y_sp])
+ax.plot(x_sp, y_sp, 'k.', markersize = 2, alpha = 0.1)
+counties.plot(ax = ax, color = 'none', edgecolor = 'dimgray')
+ax.set_title('San Francisco Bay Area - SciPy Kernel Density Estimation for Wells', fontdict = {'fontsize': '15', 'fontweight' : '3'})
+plt.show()
+```
+
+We can export the raster if necessary.
+
+```{code-cell} ipython3
+def export_kde_raster(Z, XX, YY, min_x, max_x, min_y, max_y, proj, filename):
+    '''Export and save a kernel density raster.'''
+
+    # Flip array vertically and rotate 270 degrees
+    Z_export = np.rot90(np.flip(Z, 0), 3)
+
+    # Get resolution
+    xres = (max_x - min_x) / len(XX)
+    yres = (max_y - min_y) / len(YY)
+
+    # Set transform
+    transform = Affine.translation(min_x - xres / 2, min_y - yres / 2) * Affine.scale(xres, yres)
+
+    # Export array as raster
+    with rasterio.open(
+            filename,
+            mode = "w",
+            driver = "GTiff",
+            height = Z_export.shape[0],
+            width = Z_export.shape[1],
+            count = 1,
+            dtype = Z_export.dtype,
+            crs = proj,
+            transform = transform,
+    ) as new_dataset:
+            new_dataset.write(Z_export, 1)
+```
+
+```{code-cell} ipython3
+# Export raster
+export_kde_raster(Z = Z_sp, XX = XX_sp, YY = YY_sp,
+                  min_x = min_x_sp, max_x = max_x_sp, min_y = min_y_sp, max_y = max_y_sp,
+                  proj = proj, filename = "../_static/e_vector_shapefiles/outputs/bay-area-wells_kde_scipy.tif")
+```
+
+### Method 3 - Display and export with `scikit-learn`
```

Comment:
I'm thinking we drop the scipy version since this is more comprehensive. Let's discuss the parameters in the intro to this section `bandwidth = 0.04, metric = 'euclidean', kernel = 'gaussian', algorithm = 'auto'`

Reply:
Ok, that sounds good.